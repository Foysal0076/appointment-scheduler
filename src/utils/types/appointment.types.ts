import {
  AppointmentFilters,
  AppointmentStatus,
} from '../constants/appointment.constants'

export type AppointmentStatusType =
  (typeof AppointmentStatus)[keyof typeof AppointmentStatus]

export type AppointmentFiltersType =
  (typeof AppointmentFilters)[keyof typeof AppointmentFilters]
