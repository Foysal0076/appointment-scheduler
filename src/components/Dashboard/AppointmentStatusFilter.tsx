'use client'
import { useAppointmentStatusFilter } from '@/components/Dashboard/Hooks/useAppointmentStatusFilter'
import StatusPill from '@/components/Dashboard/StatusPill'
import { AppointmentFiltersType } from '@/utils/types/appointment.types'

type Props = {
  values: AppointmentFiltersType[]
  setValues: (values: AppointmentFiltersType[]) => void
}

const AppointmentStatusFilter = ({ setValues, values }: Props) => {
  const { onClick, isActive, filters } = useAppointmentStatusFilter(
    values,
    setValues
  )
  return (
    <div className='flex flex-wrap items-center gap-4 max-md:justify-center'>
      {filters.map((status) => (
        <StatusPill
          key={status}
          status={status}
          active={isActive(status)}
          onClick={() => onClick(status)}
        />
      ))}
    </div>
  )
}

export default AppointmentStatusFilter
