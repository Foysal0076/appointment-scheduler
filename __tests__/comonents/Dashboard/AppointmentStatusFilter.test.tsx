import AppointmentStatusFilter from '@/components/Dashboard/AppointmentStatusFilter'
import { AppointmentFiltersType } from '@/utils/types/appointment.types'
import { fireEvent, render } from '@testing-library/react'

describe('AppointmentStatusFilter component', () => {
  const mockSetValues = jest.fn()
  const mockValues: AppointmentFiltersType[] = ['upcoming', 'past']

  test('renders status pills correctly', () => {
    const { getByText } = render(
      <AppointmentStatusFilter setValues={mockSetValues} values={mockValues} />
    )

    expect(getByText('upcoming')).toBeInTheDocument()
    expect(getByText('past')).toBeInTheDocument()
  })

  test('calls setValues when status pill is clicked', () => {
    const { getByTestId } = render(
      <AppointmentStatusFilter setValues={mockSetValues} values={mockValues} />
    )

    fireEvent.click(getByTestId('filter-status-pill-upcoming'))
    expect(mockSetValues).toHaveBeenCalled()

    fireEvent.click(getByTestId('filter-status-pill-past'))
    expect(mockSetValues).toHaveBeenCalled()
  })

  // snapshot
  test('matches snapshot', () => {
    const { asFragment } = render(
      <AppointmentStatusFilter setValues={mockSetValues} values={mockValues} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
