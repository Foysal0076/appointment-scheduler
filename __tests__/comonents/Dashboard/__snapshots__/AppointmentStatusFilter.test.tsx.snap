// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentStatusFilter component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-wrap items-center gap-4 max-md:justify-center"
  >
    <button
      class="select-none rounded-full border px-2 py-1 text-sm capitalize transition-colors duration-200 md:text-base border-primary-500 bg-primary-400 text-white"
      data-testid="filter-status-pill-upcoming"
    >
      upcoming
    </button>
    <button
      class="select-none rounded-full border px-2 py-1 text-sm capitalize transition-colors duration-200 md:text-base border-primary-500 bg-primary-400 text-white"
      data-testid="filter-status-pill-past"
    >
      past
    </button>
  </div>
</DocumentFragment>
`;
